### PR TITLE
Add data.gouv.fr components

### PR DESCRIPTION
**Pull request by nicolaskempf57**
Files and components in datagouv-components are converted to Typescript to improve IDE support and allow types to be exported.

Existing files and components in udata-front are converted on a case-by-case basis to improve readability and use complex types not available with the current JsDoc-like syntax.

Most of the time there is no difference between JS/TS files except the move from the old JsDoc syntax to the TS one.

Due to dependency and build updates, `entryFiles` are versioned with the package version too. This cause issues with the `burst` strategy in python using `time` in debug mode. I'm adding a new parameter `force_version`, as a workaround for now. ~~Another solution without version number is to use a manifest again. This was removed because Parcel didn't support it but vite does.~~ (Manifest were removed due to other issues so we'll stay with the version number in file name).

`PKG-INFO` is used to find the current udata-front version in both local and CI. The CI build number is used for each CI steps. In some cases, develop requirements were not already installed so `setup.py` is used for now instead of `invoke`. I think using `invoke` everywhere is better but I don't know if develop requirement were not installed for a particular reason or only because they were not needed yet. 

Closes https://github.com/etalab/data.gouv.fr/issues/1179
Closes https://github.com/etalab/data.gouv.fr/issues/1101

### CODE REVIEW

**Review by maudetes — COMMENTED** [review]
Thank you a lot for this huge work!
I have checked most functionalities impacted and it seems to work very nicely!

I'm not 100% sure about build version logic, but I think we didn't have any better yet.

File: udata_front/theme/gouvfr/datagouv-components/.env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+VITE_ADMIN_ROOT='/fr/admin/'
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/datagouv-components/.env:1]: These are default values used for the package configuration. They are used when the package is build and replaced in place by the corresponding value (they are not real server env variable outside the dev mode).

It allows verticals to use it without other configuration (that's why it references production values).

There is a `config()` function that allows package user to override them. This is what we do in `udata-front` to use values based on the server provided metadata.

File: udata_front/theme/gouvfr/datagouv-components/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "src"
+  ],
+  "module": "./dist/data-gouv-fr-components.js",
+  "types": "./dist/data-gouv-fr-components.d.ts",
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview",
+    "i18n:report": "vue-i18n-extract report --vueFiles './src/**/*.?(ts|js|vue)' --languageFiles './src/locales/*.json' --separator '^'",
+    "i18n:extract": "npm run i18n:report -- --add",
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/datagouv-components/package.json:17]: The goal is have separate files for `udata-front` and `data.gouv.fr-components`. It allows us to bundle only the package translations in the NPM package.

File: udata_front/theme/gouvfr/datagouv-components/src/locales/fr.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/datagouv-components/src/locales/fr.json:1]: Yes ! Is it better to make another project ? Or use the existing one ?

[the remark this answers is not in the document]
Comment (maudetes) [reply udata_front/theme/gouvfr/datagouv-components/src/locales/fr.json:1]: I would say use the existing one but update crowdin.yml?

File: udata_front/theme/gouvfr/assets/js/components/search/Search/Search.vue [hunk 4]
```diff
@@ -208,34 +208,45 @@
...
+import { debounce } from "../../../composables/useDebouncedRef";
+import franceWithMagnifyingGlassIcon from "../../../../../templates/svg/illustrations/france_with_magnifying_glass.svg";
+import magnifyingGlassIcon from "../../../../../templates/svg/illustrations/magnifying_glass.svg";
+import type { Dataset } from "../../../types";
+import { getAllowedExtensionsUrl } from "../../../api/resources";
+import { getLicensesUrl } from "../../../api/licenses";
+
+type Facets = {
+  organization?: string;
+  tag?: string;
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/search/Search/Search.vue:235]: This is existing in master too so I think we can to it in a separate issue to include in the next release.

File: udata_front/theme/gouvfr/assets/js/components/vanilla/accordion.js [hunk 5]
```diff
@@ -24,8 +24,6 @@ The accordion panel must have :
 ```
 */
 
-import { easing, tween, styler } from "popmotion";
-
 const togglers = document.querySelectorAll("[data-accordion-button]");
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/vanilla/accordion.js:29]: Done, I forgot to add the import after the toggleAccordion function moved to the package

File: udata_front/theme/gouvfr/datagouv-components/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (maudetes) [reply udata_front/theme/gouvfr/datagouv-components/package.json:17]: Agreed, but should `inv i18n` run at the top and `data.gouv.fr-components`'s level?

**Review by maudetes — APPROVED** [review]
I think we're all clear! Thank you for this huge work :clap:

File: .circleci/config.yml [hunk 6]
```diff
@@ -91,6 +92,29 @@ jobs:
                         -r requirements/test.pip \
                         -r requirements/develop.pip
             pip install -e .
+      - run:
+          name: Set buildno var
+          command: |
```
Comment (maudetes) [inline .circleci/config.yml:97]: Maybe add an echo at the end for logging purposes? :)

**Review by maudetes — APPROVED** [review]

File: .circleci/config.yml (hunk 6 above)
Comment (nicolaskempf57) [reply .circleci/config.yml:97]: https://app.circleci.com/pipelines/github/etalab/udata-front/2944/workflows/79e83efd-cde7-4ecc-9aa3-f2f9ad8e6ed8/jobs/15120/parallel-runs/0/steps/0-104

**Pull request merged**